x-pack/solutions/security/plugins/security_solution/docs/rfcs/detection_response/customized_rule_alert_telemetry.md

```diff
@@ -0,0 +1,273 @@
+
+### What existing tools do we have at our disposal?
+
+As mentioned earlier, we already collect prebuilt rule alert telemetry from real customers in the field. This telemetry is sanitized for performance and privacy reasons through an [allowlist](https://github.com/elastic/kibana/blob/0ad59fab8835d67b476ffbdb82850649e70ab4d9/x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/filterlists/prebuilt_rules_alerts.ts) and collected every hour by [this task](https://github.com/elastic/kibana/blob/06f5a860b6a9fdf12352a9c5934ca36a4a3311ea/x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/prebuilt_rule_alerts.ts). It's separate from the [daily snapshot telemetry](https://docs.elastic.dev/telemetry/collection/snapshot-telemetry) we also collect from clusters - that telemetry is mostly used for aggregation data.
+
+We had previously looked into using daily snapshot telemetry for this problem, but given we'd need to collect all relevant alert documents over the 24 hour window it operates in, the method proved too costly and performance intensive.
+
+### Calculating the rule diff for alert objects in snapshot telemetry
+
+Since we already have existing alert telemetry in place, it makes sense to build on top of that without having to start from scratch. Given this, we can start to look at how best to include the customized rule fields into the telemetry data. An easy way to add it would be to [enrich the sanitized alerts](https://github.com/elastic/kibana/blob/06f5a860b6a9fdf12352a9c5934ca36a4a3311ea/x-pack/solutions/security/plugins/security_solution/server/lib/telemetry/tasks/prebuilt_rule_alerts.ts#L93) with a new object listing the customized fields that we can calculate for each alert as fetched. 
+
+However, we immediately run into an issue: the rule diff calculation is an expensive one, already causing multiple performance issues in endpoints that use it. Calling it for each alert in a batch that could contain upwards of 10k alerts would likely prove extremely taxing on performance, possibly even causing timeouts from the telemetry tasks.
+
+The idea was considered to use an internal cache to store the rule assets themselves so that the most cost-intensive part of the rule diff calculation - fetching the rule assets - could be sped up, but it was determined that the size of this cache could also be prohibitively large. We would potentially have to store multiple versions for every prebuilt rule of a library that is growing every month and already contains 1300+ rules.
+
+Given these constraints, the only likely option left is to have the rule diff information already be on the alert document when fetched. This way, we could simply pass it onto the alert telemetry data like the other fields in the allowlist and filter on the relevant fields in our analytics dashboards.
+
+### An overview of the `calculateRuleDiff` function and its performance issues
```

Comment:
What happens when a user reverts a customization.  If a user customizes a field, and then later reverts, at what point will it revert and be updated it the telemetry? It seems like this function will have to run on every update, (including adding/removing fields).